src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -162,6 +162,10 @@ protected virtual PreferenceSettings ReadPreferences()
         /// <returns>new migrator instance after migration</returns>
         protected virtual DynamoMigratorBase MigrateFrom(DynamoMigratorBase sourceMigrator)
         {
+            //allows us to access some default pref values of current version before prefs are
+            //actually written to disk(firstRun).
+            var currentVersionTempPrefs = new PreferenceSettings();
```

Comment:
I am not thrilled with this solution, but I am out of bandwidth to do any other refactoring, and this lets me add a test, the alternative is no test, and I can revert the version in DynamoModel.Start after the migrator has exited. (see the previous commit).

@saintentropy @pinzart90 do you see an issue with creating temp prefs like this?

Reply:
I see... There is no good way to access the default.  If you pull it out then we might forgot to sync them later correct?

Reply:
yeah, it's probably not a big deal to add 3.0.0 inline, but theres already 2 or 3 different ways we avoid migration in various places, I'd like to avoid yet another way with a random version number that we'll forget to update.


I think this class needs a refactor / improvement